### PR TITLE
Set same shared id as eid

### PR DESCRIPTION
**Pull request by SKOCHERI**
https://github.com/prebid/Prebid.js/issues/8067

### CODE REVIEW

**Review by patmmccann — APPROVED** [review]

File: pubcid.js/src/index.js [hunk 1]
```diff
@@ -2,14 +2,25 @@ const SOURCE = 'pubcid.org';
...
 
 function collector() {
-    let value = getCookie();
-    if(value){
-        return;
-    }
-    value = generateUUID();
-    document.cookie = SOURCE + '=' + value;
+    return new Promise((resolve, reject) => {
+        let cookieValue = getCookie();
```
Comment (patmmccann) [inline pubcid.js/src/index.js]: why get the cookie first if the eids is populated? Why not just go to eids first?

Comment (SKOCHERI) [reply pubcid.js/src/index.js]: @jdwieland8282  Need your inputs here. Should we reset the value even if the cookie value already exists?

Comment (jdwieland8282) [reply pubcid.js/src/index.js]: If the sharedid in eids and cookies could be different we should preference the one we think is oldest.  If we think the two values will always be the same we should read from whichever is fastest.

Comment (SKOCHERI) [reply pubcid.js/src/index.js]: @jdwieland8282 The id value doesn't contain any date info hence we cant identify which ids are older

Comment (patmmccann) [reply pubcid.js/src/index.js]: I think we should just always prefer the eid unless it is missing, publishers would likely be surprised if a different value was ever pulled. One other thing to note, this script uses cookies, but publishers often use local storage. Publishers might not like it if they deployed this script and saw their first party cookie grow, potentially hitting cdn limits they've carefully just avoided. That being said, publishers setting the sharedid server side  are using cookies, but they are quite rare I believe (we've been trying to move to this model for a bit but it is complicated with 3500 domains).

Comment (jdwieland8282) [reply pubcid.js/src/index.js]: ok lets use `eids` then (unless missing), lets also write the sharedid to LS

Comment (SKOCHERI) [reply pubcid.js/src/index.js]: Have updated the script to read from Eids and store in LS

**Review by patmmccann — APPROVED** [review]

**Review by jlukas79 — APPROVED** [review]

**Pull request merged**